samples/command/consumer/src/main.rs

```diff
@@ -0,0 +1,136 @@
+/// Start the ambient air temperature data stream.
+///
+/// # Arguments
+/// `provider_uri` - The provider_uri.
+/// `consumer_uri` - The consumer_uri.
+fn start_send_notification_repeater(provider_uri: String, consumer_uri: String) {
+    info!("Starting the Consumer's send notification repeater.");
+    tokio::spawn(async move {
+        loop {
+            info!("Invoking the send_notification command on endpoint {}", &provider_uri);
+
+            let client_result = ProviderClient::connect(provider_uri.clone()).await;
+            if client_result.is_err() {
+                continue;
+            }
+            let mut client = client_result.unwrap();
+
+            let response_id = Uuid::new_v4().to_string();
+
+            let payload: String = String::from("The send_notification request.");
+
+            let request = tonic::Request::new(InvokeRequest {
+                entity_id: String::from(SEND_NOTIFICATION_COMMAND_ID),
+                consumer_uri: consumer_uri.clone(),
+                response_id,
+                payload,
+            });
+
+            let _response = client.invoke(request).await;
+
+            thread::sleep(time::Duration::from_millis(1000));
+        }
+    });
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Setup logging.
+    Builder::new().filter(None, LevelFilter::Info).target(Target::Stdout).init();
+
+    info!("The Consumer has started.");
+
+    // Setup the HTTP server.
+    let consumer_authority = String::from("[::1]:60010");
+    let addr: SocketAddr = consumer_authority.parse()?;
+    let consumer_impl = consumer_impl::ConsumerImpl::default();
+    let server_future =
+        Server::builder().add_service(ConsumerServer::new(consumer_impl)).serve(addr);
+
+    // Obtain the DTDL for the send_notification command.
+    info!("Sending a find_by_id request to the Digital Twin Service for the DTDL for the send_notification command.");
+    let mut client = DigitalTwinClient::connect("http://[::1]:50010").await?; // Devskim: ignore DS137138
+    let request = tonic::Request::new(FindByIdRequest {
+        entity_id: String::from(SEND_NOTIFICATION_COMMAND_ID),
+    });
+    let response = client.find_by_id(request).await?;
+    let dtdl = response.into_inner().dtdl.clone();
+    info!("Received the response for the find_by_id request. The DTDL is:\n{}", &dtdl);
+
+    info!("Parsing the DTDL.");
+    let mut parser = ModelParser::new();
+    let json_texts = vec![dtdl];
+    let model_dict_result = parser.parse(&json_texts);
+    if let Err(error) = model_dict_result {
+        panic!("Failed to parse the DTDL: {error}");
+    }
+    let model_dict = model_dict_result.unwrap();
+    info!("The DTDL parser has successfully parsed the DTDL.");
+
+    // Create the id (as a DTMI) for the send_notification command.
+    let mut send_notification_command_id: Option<Dtmi> = None;
+    create_dtmi(SEND_NOTIFICATION_COMMAND_ID, &mut send_notification_command_id);
```

Comment:
why not have create_dtmi return `Option<Dtmi>` instead of None and get rid of the second mutable param?

Reply:
Good call.  Fixed.